### DISCUSSION TITLE
@headlessui/react not working with Next.Js 13

### DISCUSSION THREAD

**Discussion by a-essawy** [tailwindlabs/headlessui#2081] [Help]
After upgrading to next.js 13 with the new `app` directory. I got this error.

```
./node_modules/@headlessui/react/dist/components/combobox/combobox.js

You're importing a component that needs useEffect. It only works in a Client Component but none of its parents are marked with "use client", so they're Server Components by default.
```

### Error in terminal
![image](https://user-images.githubusercontent.com/30594965/206742746-13de676c-c236-460e-b7bc-ef83ecad9e56.png)

### Error in localhost

![image](https://user-images.githubusercontent.com/30594965/206742372-545b116b-503f-49a8-bdbf-d829b581e858.png)

if i typed `"use client"` at `app/page.jsx` it works fine, but that removes the serves-side rendering feature.

any thought?

_thank you._

**Comment by hmaiermack** [comment]
The solution is right there in the error: Combobox doesn't support SSR as it has useEffect as a dependency. I'd just define your component with use client at the lowest level possible in your component tree.

**Reply by a-essawy** [reply]
Yes, I was hoping for another solution other than the one in the error.
Because that's means wherever I use headlessui, that means that component can't be SSR, which is the new default way of component rendering in Next 13.

**Reply by Evavic44** [reply]
Just create a separate `.js` file for the headless UI component and define the "use client" flag there and you can import it into any server component without it throwing any errors.
